src/main/java/com/pinterest/secor/uploader/S3UploadManager.java

```diff
@@ -117,11 +119,23 @@ public void refresh() {}
 
     public Handle<?> upload(LogFilePath localPath) throws Exception {
         String s3Bucket = mConfig.getS3Bucket();
-        String s3Key = localPath.withPrefix(mConfig.getS3Path()).getLogFilePath();
+        Date s3MigrationDate = null;
+        String s3Path;
+        try {
+            s3MigrationDate = new SimpleDateFormat("yyyy-MM-dd").parse(mConfig.getS3MigrationDate());
+        } catch (RuntimeException e) {
+            // optional migration date, so do nothing if it's not defined
```

Comment:
I think once you had an entry in secor.common.properties, you probably won't see that RuntimeException.